File: fn_funcs/greeter-app/func.js
const fdk=require('@fnproject/fdk');

fdk.handle(function(input){
  let name = 'From Root';
  if (input.name) {
    name = input.name;
  }
  console.log('\nInside Node Hello World function')
  return {'message': 'Hello ' + name}
})

File: fn_funcs/greeter-app/goodbye/func.go
package main

import (
	"context"
	"encoding/json"
	"fmt"
	"io"
	"log"

	fdk "github.com/fnproject/fdk-go"
)

func main() {
	fdk.Handle(fdk.HandlerFunc(myHandler))
}

type Person struct {
	Name string `json:"name"`
}

func myHandler(ctx context.Context, in io.Reader, out io.Writer) {
	p := &Person{Name: "World"}
	json.NewDecoder(in).Decode(p)
	msg := struct {
		Msg string `json:"message"`
	}{
		Msg: fmt.Sprintf("Goodbye %s", p.Name),
	}
	log.Print("Inside Go Goodbye World function")
	json.NewEncoder(out).Encode(&msg)
}
